archai/nlp/nvidia_transformer_xl/nvidia_utils/proj_adaptive_softmax.py

```diff
@@ -52,9 +52,12 @@ def __init__(self, n_token, d_embed, d_proj,
         self.cutoff_ends = [0] + self.cutoffs
         self.div_val = div_val
 
-        self.shortlist_size = self.cutoffs[0]
-        self.n_clusters = len(self.cutoffs) - 1 # number of clusters will be >= 0
-        self.head_size = self.shortlist_size + self.n_clusters
+        if self.cutoffs:
+            self.n_clusters = len(self.cutoffs) - 1 # number of clusters will be >= 0
+            self.shortlist_size = self.cutoffs[0] # TODO: remove? Never used...
+            self.head_size = self.shortlist_size + self.n_clusters # TODO: remove? Never used...
+        else:
+            self.n_clusters = 0
```

Comment:
Wondering why we need this. I think the code handled the empty cutoffs already. No?